src/en/caas.md

```diff
@@ -0,0 +1,192 @@
+Title: CAAS and Juju
+TODO:  Once 2.4 is officially released remove support Note and update status output
+       Should eventually link to CAAS-charm developer documentation
+       Update when storage becomes a Juju drivable aspect.
+       Add architectural overview/diagram once CAAS becomes stable.
+       Consider manually adding a cluster (third-party installs) via `add-cloud` and `add-credential`
+
+# CAAS and Juju
+
+CAAS is *Containers as a Service*, a cloud service that allows users to upload,
+organize, run, scale, manage, and stop containers. Combining this with Juju
+opens up new practical benefits for Juju users. Currently, Juju supports
+Kubernetes as its underlying CAAS solution.
+
+Kubernetes (often abbreviated as "k8s") provides a flexible architecture for
+managing containerised applications at scale (see the
+[Kubernetes documentation][upstream-kubernetes-docs] for more information). It
+most commonly employs Docker as its container technology.
+
+## CAAS-specific workflow
+
+Here we discuss how a building and working with a CAAS environment may differ
+from a standard Juju workflow.
+
+The only CAAS-specific Juju command is `add-k8s`, which acts as the `add-cloud`
+command for Kubernetes clouds/clusters. All other concepts and commands are
+applied in the traditional Juju manner.
+
+If the Kubernetes cluster is built with Juju itself (via a bundle) and `juju
+add-k8s` is run immediately afterwards, the contents of file `~/.kube/config`
+(if it exists) are used to add the cluster and the credentials to Juju, making
+commands `add-cloud` and `add-credential` unnecessary.
+
+User credentials can still be added by way of the `add-credential`
+or `autoload-credentials` commands. Also, at any time, the k8s CLI can be used
+to add a new user to the k8s cluster.
+
+The `add-k8s` command can be used repeatedly to set up different clusters as
+long as the contents of the configuration file has been changed accordingly.
+The KUBECONFIG environment variable is useful here as it will be honoured by
+Juju when finding the file to load.
+
+We'll demonstrate the use of the `add-k8s` command below.
+    
+## Using Kubernetes with Juju
+
+First off, a Kubernetes cluster will be required. Essentially, you will use it
+as you would any other cloud that Juju interacts with: the cluster becomes the
+backing cloud.
+
+To summarise, the steps for using Kubernetes with Juju are:
+
+ 1. Obtain a cluster
+ 1. Add the cluster to Juju
+ 1. Create a controller (and optionally a model)
```

Comment:
So this is where some of the add-k8s refinement is happening. Currently you need a controller before add-k8s is run since the command tells the controller about the cloud as well as storing the info locally. Cory's latest comment in the bug has prompted a slight rethink there.